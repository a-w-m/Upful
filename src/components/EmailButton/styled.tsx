import styled from 'styled-components'
export {Button as Button} from "../ShareButton/styled"
export {TextWrapper as TextWrapper} from "../ShareButton/styled"
export const MailToLink  = styled.a`
display: flex;
justify-content: center;
column-gap: 1.6rem;
text-decoration: none;
color: #000;
width: 100%;
border-top: 0.1rem solid var(--border-color);
border-bottom: 0.1rem solid var(--border-color);
padding: 2rem 0;
`