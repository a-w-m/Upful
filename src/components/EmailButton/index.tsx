import React from "react"
import { Email } from "../Icons/index"
import {TextWrapper, MailToLink } from "./styled"

interface EmailButtonProps {
  title: string
}

const EmailButton: React.FC<EmailButtonProps> = ({ title }) => {
  return (
    <MailToLink href={`mailto:howdy@upful.com?Subject=${title}`}>
      <Email />
      <TextWrapper>Email us about this product</TextWrapper>
    </MailToLink>
  )
}

export default EmailButton
