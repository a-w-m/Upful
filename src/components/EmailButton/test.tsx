import React from "react"
import { render, screen } from "@testing-library/react"
import EmailButton from "."

const props = {
  title: "test-email",
}

describe("Email", () => {
  beforeEach(() => {
    render(<EmailButton {...props}></EmailButton>)
  })

  it("should match snapshot", () => {
    const link = screen.getByRole("link", { name: /Email/i })
    expect(link).toMatchSnapshot()
  })

  it("should display icon", ()=>{
    const icon = screen.getByRole("img")
    expect(icon).toBeInTheDocument()
  })

  it("should display descriptive text", () => {
    const description = screen.getByText(/Email us about this product/)
    expect(description).toBeInTheDocument()
  })

  it("should have correct href", () => {
    const link = screen.getByRole("link")

    expect(link).toHaveAttribute(
      "href",
      `mailto:howdy@upful.com?Subject=${props.title}`
    )
  })
})
