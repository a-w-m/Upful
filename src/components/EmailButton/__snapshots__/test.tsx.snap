// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email should match snapshot 1`] = `
<a
  class="sc-eCYdqJ TwdGX"
  href="mailto:howdy@upful.com?Subject=test-email"
>
  <svg
    aria-label="contact email link icon"
    class="svg-icon icon-mail "
    fill="none"
    height="20"
    role="img"
    viewBox="0 0 27 22"
    width="27"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Mail Icon
    </title>
    <path
      clip-rule="evenodd"
      d="M1.458 1.5h24.084v17H1.458v-17zM0 0H27v20H0V0zm3.461 4.608l9.72 7.222.427.317.427-.317 9.72-7.222-.854-1.216-9.293 6.905-9.293-6.905-.854 1.216z"
      fill="currentColor"
      fill-rule="evenodd"
      transform="translate(0 1)"
    />
  </svg>
  <span
    class="sc-dkzDqf eWJTSj"
  >
    Email us about this product
  </span>
</a>
`;
